test-case-reporting/src/test_result_record.ts

```diff
@@ -295,4 +295,35 @@ export class TestResultRecord {
     }));
     /* eslint-enable @typescript-eslint/camelcase */
   }
+
+  /**
+   * Gets a list of the test cases with highest fail/pass/skip/error percentage
+   * @param pageInfo Object with the limit and the offset of the query
+   * @param column Which test status we want to sort by
+   * @param sampleSize The sample size of the computed stats we want
+   */
+  async getTestCasesSortedByStatColumn(
+    {limit, offset}: PageInfo,
+    column: 'passed' | 'failed' | 'skipped' | 'errored',
+    sampleSize: number
+  ): Promise<Array<TestCase>> {
+    const dbTestCases: Array<DB.TestCase> = await this.db('test_case_stats')
+      .where('test_case_stats.sample_size', sampleSize)
+      .join('test_cases', 'test_cases.id', 'test_case_stats.test_case_id')
+      .select<Array<DB.TestCase>>(
```

Comment:
The casting here seems kind of gross, but it's the only way I could see to make Typescript stop complaining, and I had trouble following the knex documentation for knex + typescript. Thoughts?

Reply:
Typescript annotations seem fine here